devdoc/interlocked_requirements.md

```diff
@@ -0,0 +1,298 @@
+
+## interlocked_decrement
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_decrement, volatile int32_t*, addend);
+```
+
+**SRS_INTERLOCKED_43_011[** `interlocked_decrement` shall decrement (decrease by one) the value of the specified 32-bit variable as an atomic operation.**]**
+
+## interlocked_decrement_16
+
+```c
+MOCKABLE_FUNCTION(, int16_t, interlocked_decrement_16, volatile int16_t*, addend);
+```
+
+**SRS_INTERLOCKED_43_012 [** `interlocked_decrement_16` shall decrement (decrease by one) the value of the specified 16-bit variable as an atomic operation.**]**
+
+
+## interlocked_decrement_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_decrement_64, volatile int64_t*, addend);
+```
+
+**SRS_INTERLOCKED_43_013 [** `interlocked_decrement_64` shall decrement (decrease by one) the value of the specified 64-bit variable as an atomic operation.**]**
+
+## interlocked_exchange
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_exchange, volatile int32_t*, target, int32_t, value);
+```
+
+**SRS_INTERLOCKED_43_014 [** `interlocked_exchange` shall set a 32-bit variable to the specified value as an atomic operation.**]**
+
+## interlocked_exchange_16
+
+```c
+MOCKABLE_FUNCTION(, int16_t, interlocked_exchange_16, volatile int16_t*, target, int16_t, value);
+```
+
+**SRS_INTERLOCKED_43_015 [** `interlocked_exchange_16` shall set a 16-bit variable to the specified value as an atomic operation.**]**
+
+
+## interlocked_exchange_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_exchange_64, volatile int64_t*, target, int64_t, value);
+```
+
+**SRS_INTERLOCKED_43_016 [** `interlocked_exchange_64` shall set a 64-bit variable to the specified value as an atomic operation.**]**
+
+## interlocked_exchange_8
+
+```c
+MOCKABLE_FUNCTION(, int8_t, interlocked_exchange_8, volatile int8_t*, target, int8_t, value);
+```
+
+**SRS_INTERLOCKED_43_017 [** `interlocked_exchange_8` shall set an 8-bit variable to the specified value as an atomic operation.**]**
+
+## interlocked_exchange_add
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_exchange_add, volatile int32_t*, addend, int32_t, value);
+```
+
+**SRS_INTERLOCKED_43_018 [** `interlocked_exchange_add` shall perform an atomic addition of two 32-bit values.**]**
+
+## interlocked_exchange_add_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_exchange_add_64, volatile int64_t*, addend, int64_t, value);
+```
+
+**SRS_INTERLOCKED_43_019 [** `interlocked_exchange_add_64` shall perform an atomic addition of two 64-bit values.**]**
+
+## interlocked_exchange_pointer
+
+```c
+MOCKABLE_FUNCTION(, void*, interlocked_exchange_pointer, volatile void*, target, void*, value);
+```
+
+**SRS_INTERLOCKED_43_020 [** `interlocked_exchange_pointer` shall atomically exchange a pair of addresses.**]**
```

Comment:
>shall atomically exchange a pair of addresses. [](start = 60, length = 46)

This says what the function sounds like it does (almost like a "swap" of the values), but really it's a "set". That is, `target` is atomically set to `value`. `value` doesn't change. I think the spec should make that clear.